Restructure NEST 3.0 guides (#1929)
nest/nest-simulator: doc/userdoc/guides/nest2_to_nest3/nest3_features/recording_simulations.rst

Recording from simulations
==========================

The ``spike_detector`` has been renamed to ``spike_recorder``. The main
rationale behind this is that the device is actually not detecting the
occurence of spikes, but rather only records them. Moreover, the new
name is more consistent with the naming of other similar devices that

> also end in the suffix `_recorder`.

Reviewer: ```suggestion
also end in `_recorder`.
```

  |     nest.SetStatus(mm, {'to_memory': True})          |     mm.record_to = "memory"        |
  |                                                      |                                    |
  +------------------------------------------------------+------------------------------------+
  |                                                      |                                    |
  | ::                                                   | ::                                 |
  |                                                      |                                    |
  |     nest.Create('spike_detector')                    |     nest.Create('spike_recorder')  |
  |                                                      |                                    |
  +------------------------------------------------------+------------------------------------+

You can retrieve the list of available backends using the following command:

 ::

    list(nest.GetKernelStatus("recording_backends").keys())

Previously, the content and formatting of any output created by a
recording device could be configured in a fine-grained fashion using

> flags like `withgid`, `withtime`, `withweight`, `withport` and so
> on. In many cases, this, however, lead to a confusing variety of

Reviewer: ```suggestion
flags like `withgid`, `withtime`, `withweight`, `withport` and some
more. In many cases, this, however, lead to a confusing variety of
```
